bin/copyRawDataToTmp.sh

```diff
@@ -0,0 +1,370 @@
+	fi
+done
+
+if [[ -z "${samplesheetsServerLocation:-}" ]]
+then
+	samplesheetsServerLocation="localhost"
+	samplesheetsLocation="${WORKING_DIR}/Samplesheets/"
+	log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "samplesheetsServerLocation set to ${samplesheetsServerLocation}."
+else
+	# shellcheck disable=SC2153
+	samplesheetsLocation="/groups/${GROUP}/${TMP_LFS}/Samplesheets/"
+fi
+
+
+#
+# Write access to prm storage requires data manager account.
+#
+if [[ "${ROLE_USER}" != "${DATA_MANAGER}" ]]; then
+	log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' "This script must be executed by user ${DATA_MANAGER}, but you are ${ROLE_USER} (${REAL_USER})."
+fi
+
+#
+# Make sure only one copy of this script runs simultaneously
+# per data collection we want to copy to prm -> one copy per group per combination of ${sourceServer} and ${SCR_ROOT_DIR}.
+# Therefore locking must be done after
+# * sourcing the file containing the lock function,
+# * sourcing config files,
+# * and parsing commandline arguments,
+# but before doing the actual data transfers.
+#
+# As servernames and folders may contain various characters that would require escaping in (lock) file names,
+# we compute a hash for the combination of ${sourceServer} and ${SCR_ROOT_DIR} to append to the ${SCRIPT_NAME}
+# for creating unique lock file. We write the combination of ${sourceServer} and ${SCR_ROOT_DIR} in the lock file
+# to make it easier to detect which combination of ${sourceServer} and ${SCR_ROOT_DIR} the lock file is for.
+#
+hashedSource="$(printf '%s:%s' "${samplesheetsServerLocation}" "${samplesheetsLocation}" | md5sum | awk '{print $1}')"
+lockFile="${WORKING_DIR}/logs/${SCRIPT_NAME}_${hashedSource}.lock"
+thereShallBeOnlyOne "${lockFile}"
+printf 'Lock file for %s instance that pushes data to %s:%s\n' "${SCRIPT_NAME}" "${samplesheetsServerLocation}" "${samplesheetsLocation}" > "${lockFile}"
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Successfully got exclusive access to lock file ${lockFile} ..."
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Log files will be written to ${WORKING_DIR}/logs ..."
+
+#
+# Use multiplexing to reduce the amount of SSH connections created
+# when rsyncing using the group's data manager account.
+# 
+#  1. Become the "${DATA_MANAGER} user who will rsync the data to prm and 
+#  2. Add to ~/.ssh/config:
+#		ControlMaster auto
+#		ControlPath ~/.ssh/tmp/%h_%p_%r
+#		ControlPersist 5m
+#  3. Create ~/.ssh/tmp dir:
+#		mkdir -p -m 700 ~/.ssh/tmp
+#  4. Recursively restrict access to the ~/.ssh dir to allow only the owner/user:
+#		chmod -R go-rwx ~/.ssh
+#
+
+#
+# Get a list of all samplesheets for this group on the specified sourceServer, where the raw data was generated, and
+#	1. Loop over their analysis ("run") sub dirs and check if there are any we need to rsync.
+#	2. Optionally, split the samplesheets per project after the data was rsynced.
+#
+# shellcheck disable=SC2029
+
```

Comment:
Remove the empty line: Any `#shellcheck disable` comments must be on the line preceeding the stuff for which you want to make an exception.